src/Microsoft.DotNet.ProjectJsonMigration/ProjectDependencyFinder.cs

```diff
@@ -104,10 +105,14 @@ public IEnumerable<ProjectDependency> ResolveProjectDependencies(ProjectContext
                 var projectExportName = projectExport.Library.Identity.Name;
                 ProjectDependency projectDependency;
 
+                if (preResolvedProjects.Contains(projectExportName))
```

Comment:
Otherwise the scenario where csproj reference present in any of the possible search paths will fail. I dont understand how this will change the preference order.